test/rest_api/test_repair_task.py

```diff
@@ -0,0 +1,68 @@
+import sys
+
+# Use the util.py library from ../cql-pytest:
+sys.path.insert(1, sys.path[0] + '/../cql-pytest')
+from util import new_test_table, new_test_keyspace
+from rest_util import set_tmp_task_ttl
+from task_manager_utils import list_tasks, drain_module_tasks, get_task_status, check_child_parent_relationship
+
+empty_id = "00000000-0000-0000-0000-000000000000"
+module_name = "repair"
+long_time = 1000000000
+
+def run_repair_and_wait(rest_api, keyspace):
+    resp = rest_api.send("POST", f"storage_service/repair_async/{keyspace}")
+    resp.raise_for_status()
+    sequence_number = resp.json()
+    resp = rest_api.send("GET", f"storage_service/repair_status", { "id": sequence_number })
+    resp.raise_for_status()
+    return sequence_number
+
+def test_repair_task_progress(cql, this_dc, rest_api):
+    drain_module_tasks(rest_api, module_name)
+    with set_tmp_task_ttl(rest_api, long_time):
+        # Insert some data.
+        with new_test_keyspace(cql, f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', '{this_dc}' : 1 }}") as keyspace:
+            schema = 'p int, v text, primary key (p)'
+            with new_test_table(cql, keyspace, schema) as t0:
+                stmt = cql.prepare(f"INSERT INTO {t0} (p, v) VALUES (?, ?)")
+                cql.execute(stmt, [0, 'hello'])
+
+                sequence_number = run_repair_and_wait(rest_api, keyspace)
+
+                # Get all repairs.
+                statuses = [get_task_status(rest_api, task["task_id"]) for task in list_tasks(rest_api, "repair") if task["sequence_number"] == sequence_number]
+                assert len(statuses) == 1, "Wrong number of internal repair tasks"
+                status = statuses[0]
+                assert status["progress_completed"] == status["progress_total"], "Incorrect task progress"
+
+                assert "children_ids" in status, "Shard tasks weren't created"
+                children = [get_task_status(rest_api, child_id) for child_id in status["children_ids"]]
+                assert all([child["progress_completed"] == child["progress_total"] for child in children]), "Some shard tasks have incorrect progress"
+
+                assert sum([child["progress_total"] for child in children]) == status["progress_total"], "Total progress of parent is not equal to children total progress sum"
+                assert sum([child["progress_completed"] for child in children]) == status["progress_completed"], "Completed progress of parent is not equal to children completed progress sum"
+    drain_module_tasks(rest_api, module_name)
+
+def check_repair_task_tree(cql, this_dc, rest_api):
+    drain_module_tasks(rest_api, module_name)
+    with set_tmp_task_ttl(rest_api, long_time):
+        with new_test_keyspace(cql, f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', '{this_dc}' : 1 }}") as keyspace:
+            schema = 'p int, v text, primary key (p)'
+            with new_test_table(cql, keyspace, schema) as t0:
```

Comment:
Can we have more than one table in the test to make sure we handle multiple table ranges correclty?